### DISCUSSION TITLE
Virtual background using Ant Media Server REST API

### DISCUSSION THREAD

**Discussion by deninom** [ant-media/Ant-Media-Server#6765] [Q&A]
Is it possible to enable the virtual background using Ant Media Server REST API instead of Javascript SDK?
In this case how can I do it?

**Comment by lastpeony** [comment]
No because its working on client side.

You can build your own logic by connecting all clients through websocket to your application server, then triggering virtual background event through that.

**Comment by rahul78275** [comment]
Hello @deninom

Can you please share any use case where virtual background required to be implemented using rest api.

**Reply by deninom** [reply]
Yes of course. A typical scenario is the case where you have your own camera and yo want replace the background with another image that you can pick from your local disk. This is possible using AntMedia javascript SDK and there is also an example provided by Ant Media. In my case, I cannot use the javascript SDK so I am trying to do it in a different way and the REST API is one of these options. 
If you have any other idea, please share it with me. Any suggestion is accepted.

**Reply by rahul78275** [reply]
No idea @deninom. But this solution will consume more CPU on server side.

**Reply by burak-58** [reply]
If you have a single color background, you can do it by Filter Plugin as told [here](https://antmedia.io/discover-the-filter-plugin/). Otherwise it requires some AI algorithm to detect background and there is no direct solution in backend for this.

Why don't you prefer doing it with JS SDK?

**Reply by deninom** [reply]
I have a windows client application that acquire and send the video stream to Ant Media Server so I cannot use JS SDK.  The filter plugin can be an idea and I need to check it . As an alternative I need to add an AI algorithm on client application.

**Reply by burak-58** [reply]
We have one more possibility: [Media-Push plugin](https://github.com/ant-media/Plugins/tree/master/MediaPushPlugin).

Shortly, it can capture any webpage and publish it to AMS as WebRTC stream. You can create a page (by extending AMS play.html) which apply AI alogrithm then publish it to AMS. It will be a server side solution also.
